creme/time_series/exponential_smoothing.py

```diff
@@ -0,0 +1,566 @@
+            self.s = collections.deque([0 for _ in range(m)], maxlen=m)
+        else:
+            self.s = collections.deque(s, maxlen=m)
+
+        self.lt = l0
+        self.bt = b0
+
+    def fit_one(self, y: float):
+
+        st_1 = self.s[-1]
+        st_m = self.s[-self.m]
+
+        lt_1 = self.lt
+        bt_1 = self.bt
+
+        self.lt = self.alpha * (y - st_m) + (1 - self.alpha) * (lt_1 + bt_1)
+        self.bt = self.beta * (self.lt - lt_1) + (1 - self.beta) * bt_1
+        st = self.gamma * (y - lt_1 - bt_1) + (1 - self.gamma) * st_m
+        self.s.append(st)
+
+        return self
+
+    def forecast(self, horizon: int) -> list:
+        return [
+            self.lt + h * self.bt + self.s[h - self.m * ((
+                (h - 1) // self.m) + 1)] for h in range(1, horizon + 1)
+        ]
+
+
+class HoltWinterMultiplicative(base.Forecaster):
+    """
+    #TODO : Docstring + test
+
+    Parameters:
+        m (int): m the frequency of the seasonality
+        alpha (float): The smoothing parameter for the level, 0 ≤ alpha ≤ 1 . Defaults to `0.5`.
+        beta (float): The smoothing parameter for the trend, 0 ≤ beta ≤ 1 . Defaults to `0.5`.
+        gamma (float): The smoothing parameter seasonal component 0 ≤ gamma ≤ 1. Defaults to `0.5`.
+        s (List[float]): Initialization values for the seasonality. Defaults to `None`.
+        l0 (float): Initialization value for the level. Defaults to `0.5`.
+        b0 (float): Initialization value for the trend. Defaults to `0.5`.
+
+    Example:
+    ::
+        >>> #TODO
+
+    References:
+        1. `Holt-Winters’ multiplicative method <https://otexts.com/fpp2/holt-winters.html>`_
+    """
+
+    def __init__(
+            self,
+            m: int,
+            alpha: float = 0.5,
+            beta: float = 0.5,
+            gamma: float = 0.5,
+            s: Optional[List[float]] = None,  # We can discuss for this choice
+            l0: float = 0.5,
+            b0: float = 0.5):
+
+        self.m = m
+
+        self.alpha = alpha
+        self.beta = beta
+        self.gamma = gamma
+
+        # init s : [s_{t-m}, s_{t-m+1}, ..., s_{t}]
+        if s is None:
+            self.s = collections.deque([1 for _ in range(m)], maxlen=m)
+        else:
+            self.s = collections.deque(s, maxlen=m)
+
+        self.lt = l0
+        self.bt = b0
+
+    def fit_one(self, y: float):
+
+        st_1 = self.s[-1]
+        st_m = self.s[-self.m]
+
+        lt_1 = self.lt
+        bt_1 = self.bt
+
+        # replace by safe division
+        self.lt = self.alpha * (y / st_m) + (1 - self.alpha) * (lt_1 + bt_1)
+        self.bt = self.beta * (self.lt - lt_1) + (1 - self.beta) * bt_1
+        st = self.gamma * (y / (lt_1 + bt_1)) + (1 - self.gamma) * st_m
+        self.s.append(st)
+
+        return self
+
+    def forecast(self, horizon: int) -> list:
+        return [(self.lt + h * self.bt) * self.s[h - self.m * ((
+            (h - 1) // self.m) + 1)] for h in range(1, horizon + 1)]
```

Comment:
Can you break this into multiple lines? For instance:

```py
return [
    (self.lt + h * self.bt) * self.s[h - self.m * (((h - 1) // self.m) + 1)]
    for h in range(1, horizon + 1)
]
```

In general I'm in favor of writing list comprehensions over multiple lines, I find that it increases readability. As you wish though! If you do change this, then please change all the other list comprehensions in your code :)